sdk/storage/storage-blob/review/generated/src/models/index.d.ts

```diff
@@ -0,0 +1,6713 @@
+    /**
+     * Indicates whether metrics should generate summary statistics for called API operations.
+     */
+    includeAPIs?: boolean;
+    retentionPolicy?: RetentionPolicy;
+}
+/**
+ * An interface representing PageRange.
+ */
+export interface PageRange {
+    start: number;
+    end: number;
+}
+/**
+ * An interface representing ClearRange.
+ */
+export interface ClearRange {
+    start: number;
+    end: number;
+}
+/**
+ * the list of pages
+ */
+export interface PageList {
+    pageRange?: PageRange[];
+    clearRange?: ClearRange[];
+}
+/**
+ * signed identifier
+ */
+export interface SignedIdentifier {
+    /**
+     * a unique id
+     */
+    id: string;
+    accessPolicy: AccessPolicy;
+}
+/**
+ * The properties that enable an account to host a static website
+ */
+export interface StaticWebsite {
+    /**
+     * Indicates whether this account is hosting a static website
+     */
+    enabled: boolean;
+    /**
+     * The default name of the index page under each directory
+     */
+    indexDocument?: string;
+    /**
+     * The absolute path of the custom 404 page
+     */
+    errorDocument404Path?: string;
+}
+/**
+ * Storage Service Properties.
+ */
+export interface StorageServiceProperties {
+    logging?: Logging;
+    hourMetrics?: Metrics;
+    minuteMetrics?: Metrics;
+    /**
+     * The set of CORS rules.
+     */
+    cors?: CorsRule[];
+    /**
+     * The default version to use for requests to the Blob service if an incoming request's version
+     * is not specified. Possible values include version 2008-10-27 and all more recent versions
+     */
+    defaultServiceVersion?: string;
+    deleteRetentionPolicy?: RetentionPolicy;
+    staticWebsite?: StaticWebsite;
+}
+/**
+ * Stats for the storage service.
+ */
+export interface StorageServiceStats {
+    geoReplication?: GeoReplication;
+}
+/**
+ * Additional parameters for a set of operations.
+ */
+export interface LeaseAccessConditions {
+    /**
+     * If specified, the operation only succeeds if the resource's lease is active and matches this
+     * ID.
+     */
+    leaseId?: string;
+}
+/**
+ * Additional parameters for a set of operations.
+ */
+export interface ModifiedAccessConditions {
+    /**
+     * Specify this header value to operate only on a blob if it has been modified since the
+     * specified date/time.
+     */
+    ifModifiedSince?: Date;
+    /**
+     * Specify this header value to operate only on a blob if it has not been modified since the
+     * specified date/time.
+     */
+    ifUnmodifiedSince?: Date;
+    /**
+     * Specify an ETag value to operate only on blobs with a matching value.
+     */
+    ifMatch?: string;
+    /**
+     * Specify an ETag value to operate only on blobs without a matching value.
+     */
+    ifNoneMatch?: string;
+}
+/**
+ * Additional parameters for a set of operations, such as: Directory_create, Directory_rename,
+ * Blob_rename.
+ */
+export interface DirectoryHttpHeaders {
```

Comment:
Why DirectoryHttpHeaders? These HTTP headers can be applied to a blob

Reply:
Yes they share most of the headers, although blob header has [an extra field `blobContentMD5`](https://github.com/Azure/azure-sdk-for-js/blob/feature%2Fstorage/sdk/storage/storage-blob/src/generated/src/models/index.ts#L645)

Reply:
JS users don't use these types directly very much. The structure of the types are more important. As these are generated code, we think it adds little value to have a shared base type then a more specific one for blob http headers.